scriptworker/cot/verify.py

```diff
@@ -483,15 +501,9 @@ def find_task_dependencies(task, name, task_id):
     log.info("build_task_dependencies {} {}".format(name, my_task_id))
     if name.count(':') > 5:
         raise CoTError("Too deep recursion!\n{}".format(name))
-    deps = find_task_dependencies(task, name, my_task_id)
-    task_names = sorted(deps.keys())
-    # make sure we deal with the decision task first, or we may populate
-    # signing:build0:decision before signing:decision
-    decision_key = "{}:decision".format(name)
-    if decision_key in task_names:
-        task_names = [decision_key] + sorted([x for x in task_names if x != decision_key])
-    for task_name in task_names:
-        task_id = deps[task_name]
+    sorted_dependencies = find_sorted_task_dependencies(task, name, my_task_id)
+
+    for task_name, task_id in sorted_dependencies:
         if task_id not in chain.dependent_task_ids():
```

Comment:
Evidently this line gets rid of `task_id` dups, so filtering out duplicates is good but not required :)